lib/kennel.rb

```diff
@@ -54,48 +54,88 @@ class << self
   self.err = $stderr
 
   class Engine
-    attr_accessor :strict_imports
+    def initialize(
+      update_generated: true,
+      show_plan: false,
+      require_confirm: nil,
+      update_datadog: false,
+      strict_imports: true
```

Comment:
feels a bit strange that actions and options are on the same level

Reply:
The option to run an action. 🤷‍♀️

Reply:
afaik that's not a very common pattern, usually there are options for how a thing behaves and then methods are called to perform actions
... I get why it's needed though

Reply:
* `Kennel::Engine.new(**some_options).run`
* `Kennel::Engine.new(**some_options).run(**more_options)`

are equally good/bad IMO. Either way will still have the constraint that the only useful pattern is `.new` followed by `#run` (exactly once).

One could go with more of a 1:1 with the rake tasks:

* Kennel::Engine.new(**some_options).generate`
* Kennel::Engine.new(**some_options).plan`
* Kennel::Engine.new(**some_options).update_datadog`

but (a) that only really makes sense insofar as `rake plan` is actually doing a `parts` and `generate` and `syncer` and then returning the syncer plan, i.e. it's dependent on the internals of those tasks. And (b) we already diverge significantly from this with our `ci_and_notify` or whatever it's called.
 
The existing code already allows for:
 * build parts
 * build parts, save to `generated`
 * build parts, show the plan
 * build parts, save to `generated`, show the plan
 * build parts, show the plan, update datadog
 * build parts, save to `generated`, show the plan, update datadog
 * build parts, show the plan, get confirmation, update datadog
 * build parts, save to `generated`, show the plan, get confirmation, update datadog

so this PR just exposes what's already going on more clearly (one could argue).

Reply:
yeah, good point